### PR TITLE
tailcfg: define a type for NodeCapability

### PR DESCRIPTION
**Pull request by maisem**
Instead of untyped string, add a type to identify these.

Updates #cleanup

### CODE REVIEW

File: tailcfg/tailcfg.go [hunk 1]
```diff
@@ -1221,11 +1222,22 @@ type CapGrant struct {
...
-// It's a string, but its meaning is application-defined.
-// It must be a URL, like "https://tailscale.com/cap/file-sharing-target" or
-// "https://example.com/cap/read-access".
-type PeerCapability string
+// CapabilityName is a string used to identify a specific capability. Its
+// meaning is application-defined.
+//
+// It must be a URL like "https://tailscale.com/cap/file-send", or a well-known
+// capability name like 'funnel'.
+type CapabilityName string
```
Comment (bradfitz) [inline tailcfg/tailcfg.go:1230]: Is there any reason for this? why not just make the other two be of type string. there's no inheritance going on here.

what would use this type?

File: tailcfg/tailcfg.go [hunk 2]
```diff
@@ -1221,11 +1222,22 @@ type CapGrant struct {
...
+
+// NodeCapability is a capability granted to a node.
+//
+// Unlike PeerCapability, NodeCapability is not in context of a peer and is
+// granted to the node itself.
+type NodeCapability CapabilityName
```
Comment (bradfitz) [inline tailcfg/tailcfg.go:1236]: trying to avoid "node attribute"? don't want to use that name or at least mention it as another name for this?

**Review by creachadair — APPROVED** [review]

File: ipn/serve_test.go [hunk 3]
```diff
@@ -9,20 +9,20 @@ import (
...
-		{8091, []string{portAttr, tailcfg.CapabilityHTTPS, tailcfg.NodeAttrFunnel}, true},
-		{3000, []string{portAttr, tailcfg.CapabilityHTTPS, tailcfg.NodeAttrFunnel}, true},
+		{443, []tailcfg.NodeCapability{portAttr}, true}, // No "funnel" attribute
+		{443, []tailcfg.NodeCapability{portAttr, tailcfg.NodeAttrFunnel}, true},
+		{443, []tailcfg.NodeCapability{portAttr, tailcfg.CapabilityHTTPS, tailcfg.NodeAttrFunnel}, false},
+		{8443, []tailcfg.NodeCapability{portAttr, tailcfg.CapabilityHTTPS, tailcfg.NodeAttrFunnel}, false},
+		{8321, []tailcfg.NodeCapability{portAttr, tailcfg.CapabilityHTTPS, tailcfg.NodeAttrFunnel}, true},
+		{8083, []tailcfg.NodeCapability{portAttr, tailcfg.CapabilityHTTPS, tailcfg.NodeAttrFunnel}, false},
+		{8091, []tailcfg.NodeCapability{portAttr, tailcfg.CapabilityHTTPS, tailcfg.NodeAttrFunnel}, true},
+		{3000, []tailcfg.NodeCapability{portAttr, tailcfg.CapabilityHTTPS, tailcfg.NodeAttrFunnel}, true},
```
Comment (bradfitz) [inline ipn/serve_test.go:25]: make a local `type cap = tailcfg.NodeCapability` to make these less visually noisy?

or `caps := func(x ....tailcfg.NodeCapability) []tailcfg.NodeCapability { return x }` ?

File: tailcfg/tailcfg.go (hunk 1 above)
Comment (maisem) [reply tailcfg/tailcfg.go:1230]: this was only for documentation purposes, can remove

File: tailcfg/tailcfg.go [hunk 4]
```diff
@@ -1911,112 +1913,116 @@ type Oauth2Token struct {
...
-	// These are the capabilities that the self node has as listed in
-	// MapResponse.Node.Capabilities.
-	//
-	// We've since started referring to these as "Node Attributes" ("nodeAttrs"
-	// in the ACL policy file).
+// NodeCapability represents a capability granted to the self node as listed in
+// MapResponse.Node.Capabilities.
+//
+// It must be a URL like "https://tailscale.com/cap/file-sharing", or a
+// well-known capability name like 'funnel'.
```
Comment (bradfitz) [inline tailcfg/tailcfg.go:1920]: double quotes here like you did in other string

Comment (bradfitz) [inline tailcfg/tailcfg.go:1920]: mention that this well-known short form is restricted to Tailscale-official things and aren't available for end users?

**Review by bradfitz — APPROVED** [review]

**Pull request merged**